src/cargo/util/mod.rs

```diff
@@ -73,6 +73,12 @@ pub fn elapsed(duration: Duration) -> String {
     }
 }
 
+pub fn format_bytes(bytes: f32) -> (f32, &'static str) {
```

Comment:
> It may make more sense accepting `usize` instead of `f32` IMO.

This function was originally used for a _rate_ estimate, so the input was an average number of bytes over a period of a minute. In that use case, the input could have been a non-integer number of bytes. But I agree that `usize` makes more sense here; I can round the input at that original callsite.

> I cannot recall the exact reason not including B in UNITS, but it may be better to have a B suffix

I think it's a micro-optimization? i.e. the `B` is constant, so only return the part that might change. I can change that too since it's a fairly trivial thing, may as well improve ergonomics since it will be used in multiple places now.

Reply:
Ok, took all your suggestions (although I left `ZiB` and `YiB` out of `UNITS`, since `usize::MAX` is already only `16 EiB`). And I added some tests as well.

Reply:
Just realized that `usize` might be only 4GiB maximum on 32-bits machines. Although there is a hard limit (500MiB) for unpack crate at this moment, I still feel like Cargo should handle packing project larger than `u32::MAX`. What do you think about changing this to `u64` explicitly? 

I believe this is the last thing to check before merging this :)

Reply:
That makes sense, updated :smile:

Reply:
> Edit: one more thing to note - the calculation requires `bytes` to be cast to an `f32` anyways, but that could be improved slightly once we have [rust-lang/rust#70887](https://github.com/rust-lang/rust/issues/70887) stabilized.

I believe we could do this today on 1.67 :D